/*

***********************************************************************************
* Copyright (C) 2018 - 2020, BlockSettle AB
* Distributed under the GNU Affero General Public License (AGPL v3)
* See LICENSE or http://www.gnu.org/licenses/agpl.html
*
**********************************************************************************

*/
syntax = "proto3";

package Blocksettle.Storage.Signer;


enum FileType
{
   RequestFileType   = 0;
   SignedTXFileType  = 1;
}

message FileContainer
{
   FileType	type = 1;
   bytes	data = 2;
}

message File
{
   repeated FileContainer	payload = 1;
}

message SignedTX
{
   bytes    transaction = 1;
   string   comment = 2;
   bool     allow_broadcasts = 3; // Copied from TXSignRequest::allowBroadcasts
}
